tests/test_processing_stack.py

```diff
@@ -0,0 +1,117 @@
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def detect_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+    logger.info("Detected State Machines: %s", state_machines_detection_response)
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {}
+        for tag in tags_detection_response["tags"]:
+            state_machine_tags[tag["key"]] = tag["value"]
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+            break
+
+    return datalake_state_machine
+
+
+@mark.timeout(1200)
+@mark.infrastructure
+def test_should_successfully_run_dataset_version_creation_process(
+    batch_client, stepfunctions_client
+):
+
+    state_machine_input = """{
+        "dataset_id":"1xyz",
+        "version_id":"2xyz",
+        "type":"RASTER",
+        "metadata_url":"s3://linz-geospatial-data-lake-test-data/tasman_rural_2003_1m/collection.json"
+    }"""
+
+    # speed-up Compute Environment launch
+    datalake_compute_environment = detect_compute_environment(batch_client)
+    launch_compute_environment_resources(batch_client, datalake_compute_environment)
+
+    # launch State Machine
+    datalake_state_machine = detect_state_machine(stepfunctions_client)
+    execution_response = stepfunctions_client.start_execution(
+        stateMachineArn=datalake_state_machine["stateMachineArn"], input=state_machine_input
+    )
+    logger.info("Executed State Machine: %s", execution_response)
+
+    # poll for State Machine State
+    machine_status = None
+    while machine_status in (None, "RUNNING"):
+        logger.info("%s Polling for State Machine state %s", ">" * 20, "<" * 20)
+
+        state_machine_status_response = stepfunctions_client.describe_execution(
+            executionArn=execution_response["executionArn"]
+        )
+        logger.info(
+            "State Machine Name: %s, Status: %s",
+            state_machine_status_response["name"],
+            state_machine_status_response["status"],
+        )
+        logger.info("State Machine Input: %s", state_machine_status_response.get("input"))
+        logger.info("State Machine Output: %s", state_machine_status_response.get("output"))
+
+        machine_status = state_machine_status_response["status"]
+        time.sleep(5)
+
+    assert state_machine_status_response["status"] == "SUCCEEDED"
```

Comment:
Could we simplify the loop to maybe

```
while (execution := stepfunctions_client.describe_execution(executionArn=execution_response["executionArn"]))["status"] == "RUNNING":
    sleep(5)
assert execution["status"] == "SUCCEEDED", execution
```

? The logging is useful while building the test, but it's going to make the test runs much more verbose and a bit slower.

Reply:
Done in 29ca3fa32158cf6b4da381cfd5a7120ff541aae5 , thanks.